src/mame/misc/orientp.cpp

```diff
@@ -0,0 +1,179 @@
+// license:BSD-3-Clause
+// copyright-holders:flama12333
+/*************************************************************************
+
+// Features Notes Was from archived websited:
+src: Chang yu website
+
+Pearl in the  East 
+
+5 balls per game ... win a bonus if 4 balls flip into the same slot.
+
+When balls flip into certain slots, additional lamps illuminate.
+
+When scoring on all letters of pearl in the east, up to 4 extra lamps randomly illuminate.
+
+Higher bets when scoring on all letters of PEARL IN THE EAST win a higher bonus.
+
+New feature: buy a 6th ball toup the odds.
+
+Oriental Pearl - 1997
+
+5 balls per game and 2 win modes.
+
+1 higher bets mean more bonus lights and more chances to win.
+
+Big bonus for scoring on all letters of "ORIENTAL PEARL".
+
+Electronic ball-checking device ensures where ball lands.
+
+// Hardware info - may not accurate
+the dump was from Soccer Santiago II 6 ball pinball
+
+Buttons
+K1
+K2
+K3
+K4
+
+ic
+u1 kc8279
+u17 and u21 nec d8255ac-2
+u32 hm6264
+??  winbond w27c020? adpcm rom.
+u33 winbond w27c512 boot rom.
+u39 at89s51 second mcu for protection.
+
+// TODO:
+Need hardware info.
+Hook up nvram inputs opll and adpcm.
+east8 Has undumped mcu and adpcm rom.
+Need Layout as and Add segment display as marywu.cpp
+*/
+
+#include "emu.h"
+
+#include "cpu/mcs51/mcs51.h"
+#include "machine/i8255.h"
+#include "machine/i8279.h"
+#include "sound/ay8910.h"
+#include "sound/ymopl.h"
+#include "sound/okim6295.h"
+#include "speaker.h"
+
+namespace {
+
+class orientalpearl_state : public driver_device
+{
+public:
+	orientalpearl_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+	{ }
+
+	void orientp(machine_config &config);
+
+private:
+	void io_map(address_map &map);
+	void program_map(address_map &map);
+
+protected:
```

Comment:
please put the protected section above the private section. Also, if you don't need the machine_start() method, you can delete it entirely, unless you expect you will need it later on